### ISSUE TITLE
error while loading shared libraries: libavfilter.so.5: cannot open shared object file: No such file or director

### ISSUE THREAD

**Issue by gesus14** [ejurgensen/forked-daapd#481]
I am using forked-daapd on a raspberyy-pi (osmc/debian). After pulling the latest changes via git and compiling/installing forked-daapd I get an error message on starting forked-daapd:

```
error while loading shared libraries: libavfilter.so.5: cannot open shared object file: No such file or director
```

Do you have an idea which dependencies are missing here?

Thanks in advance :-)

**Comment by ejurgensen** [comment]
Not sure how you could end in this state if it compiled ok, but my suggestion would be to clean up and try again. Remove all libav packages, e.g. with:
```
sudo apt-get -y autoremove ffmpeg
sudo apt-get -y autoremove libav-tools
sudo apt-get -y autoremove libavformat-dev libavcodec-dev libavutil-dev libswscale-dev libavfilter-dev

```
Then check for leftover stuff in /usr with "find -name libavfilter*" and get rid of that. Then install ffmpeg or libav-tools again with the -dev packages, goto the forked-daapd build dir, "make clean", autoreconf -vi, configure, make and install.

Another option is to try installing from my RPi repo, even though it is for Raspbian. Maybe it works?

**Comment by gesus14** [comment]
Thank you for your quick response. I tried both: cleaning up and recompiling as described in your comment and installing from your repo.

After cleaning up and recompiling, now I get a compilation error

```
/bin/bash ../libtool  --tag=CC   --mode=link gcc  -g -O2   -o forked-daapd main.o db.o db_init.o db_upgrade.o logger.o conffile.o cache.o library/filescanner.o library/filescanner_ffmpeg.o library/filescanner_playlist.o library/filescanner_smartpl.o library/filescanner_itunes.o library.o mdns_avahi.o remote_pairing.o avio_evbuffer.o httpd.o httpd_rsp.o httpd_daap.o httpd_dacp.o httpd_jsonapi.o httpd_streaming.o httpd_oauth.o http.o dmap_common.o transcode.o artwork.o misc.o misc_json.o rng.o rsp_query.o daap_query.o player.o worker.o input.o inputs/file_http.o inputs/pipe.o outputs.o outputs/raop.o outputs/raop_verification.o outputs/streaming.o outputs/dummy.o outputs/fifo.o outputs/alsa.o outputs/pulse.o  evrtsp/rtsp.o   mpd.o listener.o commands.o websocket.o  RSPLexer.o RSPParser.o RSP2SQL.o DAAPLexer.o DAAPParser.o DAAP2SQL.o SMARTPLLexer.o SMARTPLParser.o SMARTPL2SQL.o -lavformat -lavcodec -lswscale -lavutil -lavfilter -lgpg-error -lgcrypt -lantlr3c -levent -lmxml -lpthread -lconfuse -lz  -lavahi-common -lavahi-client -ljson-c -levent_pthreads -levent -lplist -lsodium -lwebsockets -lcurl -lpulse -lasound  -lsqlite3 -lunistring  -lpthread -lrt 
libtool: link: gcc -g -O2 -o forked-daapd main.o db.o db_init.o db_upgrade.o logger.o conffile.o cache.o library/filescanner.o library/filescanner_ffmpeg.o library/filescanner_playlist.o library/filescanner_smartpl.o library/filescanner_itunes.o library.o mdns_avahi.o remote_pairing.o avio_evbuffer.o httpd.o httpd_rsp.o httpd_daap.o httpd_dacp.o httpd_jsonapi.o httpd_streaming.o httpd_oauth.o http.o dmap_common.o transcode.o artwork.o misc.o misc_json.o rng.o rsp_query.o daap_query.o player.o worker.o input.o inputs/file_http.o inputs/pipe.o outputs.o outputs/raop.o outputs/raop_verification.o outputs/streaming.o outputs/dummy.o outputs/fifo.o outputs/alsa.o outputs/pulse.o evrtsp/rtsp.o mpd.o listener.o commands.o websocket.o RSPLexer.o RSPParser.o RSP2SQL.o DAAPLexer.o DAAPParser.o DAAP2SQL.o SMARTPLLexer.o SMARTPLParser.o SMARTPL2SQL.o  -lavformat -lavcodec -lswscale -lavutil -lavfilter /usr/lib/arm-linux-gnueabihf/libgpg-error.so -lgcrypt -lantlr3c -lmxml -lconfuse -lz -lavahi-common -lavahi-client -ljson-c -levent_pthreads -levent -lplist -lsodium -lwebsockets /usr/lib/arm-linux-gnueabihf/libcurl-gnutls.so -lpulse -lasound /usr/lib/arm-linux-gnueabihf/libsqlite3.so -lunistring -lpthread -lrt -pthread
/usr/bin/ld: /usr/lib/gcc/arm-linux-gnueabihf/6/../../../arm-linux-gnueabihf/libavfilter.a(vf_frei0r.o): undefined reference to symbol 'dlsym@@GLIBC_2.4'
//lib/arm-linux-gnueabihf/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
Makefile:705: recipe for target 'forked-daapd' failed
make[3]: *** [forked-daapd] Error 1
make[3]: Leaving directory '/home/osmc/forked-daapd/src'
Makefile:559: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/home/osmc/forked-daapd/src'
Makefile:623: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/osmc/forked-daapd'
Makefile:465: recipe for target 'all' failed
make: *** [all] Error 2
```

Unfortunately installing from your repo doesn't work either. The system service fails to start and running it from the bash leads to

```
osmc@lilo:~$ forked-daapd
forked-daapd: error while loading shared libraries: libavfilter.so.6: cannot open shared object file: No such file or directory
```

Do you have any idea how I get forked-daapd running? Thank you in advance :-)

**Comment by gesus14** [comment]
Okay, meanwhile I think, I've found a solution to get forked-daapd running by using your repository. The search for `libavfilter.so.6` lead me to two packages: `libavfilter6`and `libavfilter-extra6`. `libavfilter6` was installed. I uninstalled it and installed `libavfilter-extra6` instead. After that, forked-daapd started. :-) 

Thanks for answering my questions and helping me. I really appreciate your work on forked-daapd! It is great software :-)
